doxia-core/src/main/java/org/apache/maven/doxia/macro/MacroRequest.java

```diff
@@ -40,15 +40,23 @@ public class MacroRequest {
     /** A map of parameters. */
     private Map<String, Object> parameters;
 
+    private final Parser currentParser;
     /**
      * <p>Constructor for MacroRequest.</p>
      *
+     * @param currentParser a {@link org.apache.maven.doxia.parser.Parser} object (which is the parser triggering this macro).
      * @param sourceContent a {@link java.lang.String} object.
-     * @param parser a {@link org.apache.maven.doxia.parser.AbstractParser} object.
+     * @param parser a new {@link org.apache.maven.doxia.parser.AbstractParser} object acting as secondary parser.
      * @param param a {@link java.util.Map} object.
      * @param basedir a {@link java.io.File} object.
      */
-    public MacroRequest(String sourceContent, AbstractParser parser, Map<String, Object> param, File basedir) {
+    public MacroRequest(
+            Parser currentParser,
```

Comment:
This is exactly the idea I had, but rejected it because the I consider a macro something which should not push back, i.e., side effect free.